gtk4/src/subclass/scale.rs

```diff
@@ -0,0 +1,61 @@
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Cast;
+
+use super::range::RangeImpl;
+use crate::{Range, Scale};
+
+pub trait ScaleImpl: ScaleImplExt + RangeImpl {
+    fn get_layout_offsets(&self, scale: &Self::Type) -> Option<(i32, i32)> {
+        self.parent_get_layout_offsets(scale)
+    }
+}
+
+pub trait ScaleImplExt: ObjectSubclass {
+    fn parent_get_layout_offsets(&self, scale: &Self::Type) -> Option<(i32, i32)>;
+}
+
+impl<T: ScaleImpl> ScaleImplExt for T {
+    fn parent_get_layout_offsets(&self, scale: &Self::Type) -> Option<(i32, i32)> {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkScaleClass;
+            if let Some(f) = (*parent_class).get_layout_offsets {
+                let x = std::ptr::null_mut();
+                let y = std::ptr::null_mut();
+                f(scale.unsafe_cast_ref::<Scale>().to_glib_none().0, x, y);
+                if !x.is_null() && !y.is_null() {
+                    return Some((x as i32, y as i32));
+                }
+            }
+            None
+        }
+    }
+}
+
+unsafe impl<T: ScaleImpl> IsSubclassable<T> for Scale {
+    fn override_vfuncs(class: &mut glib::Class<Self>) {
+        <Range as IsSubclassable<T>>::override_vfuncs(class);
+
+        let klass = class.as_mut();
+        klass.get_layout_offsets = Some(scale_get_layout_offsets::<T>);
+    }
+}
+
+unsafe extern "C" fn scale_get_layout_offsets<T: ScaleImpl>(
+    ptr: *mut ffi::GtkScale,
+    x_ptr: *mut libc::c_int,
+    y_ptr: *mut libc::c_int,
+) {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<Scale> = from_glib_borrow(ptr);
+
+    if let Some((x, y)) = imp.get_layout_offsets(wrap.unsafe_cast_ref()) {
```

Comment:
Not nullable but you can always assign to the `x_ptr` / `y_ptr`. They're not optional from what I see in the C code.

Reply:
Per the docs 
>  * If the #GtkScale:draw-value property is %FALSE, the return values are undefined.
So I'm not sure if we can always to the x_ptr/y_ptr?

Reply:
```cpp
void 
gtk_scale_get_layout_offsets (GtkScale *scale,
                              int      *x,
                              int      *y)
{
  int local_x = 0; 
  int local_y = 0;

  g_return_if_fail (GTK_IS_SCALE (scale));

  if (GTK_SCALE_GET_CLASS (scale)->get_layout_offsets)
    (GTK_SCALE_GET_CLASS (scale)->get_layout_offsets) (scale, &local_x, &local_y);

  if (x)
    *x = local_x;
  
  if (y)
    *y = local_y;
}
```

is the only caller, and that assumes that some value is always returned and it always passes non-`NULL` pointers to the vfunc

Reply:
That also means that the function is optional and `0`/`0` is the default, btw.

Reply:
I somehow completely missed the ` if (GTK_SCALE_GET_CLASS (scale)->get_layout_offsets)`, thanks!